### ISSUE TITLE
jax.scipy.special.owens_t — native Owen's T function

### ISSUE THREAD

**Issue by TheSkyentist** [jax-ml/jax#37463]
This is a follow-up to #12888, which requested Owen's T for computing the Skew Normal CDF. At the time, the suggested workaround was `tfp.math.owens_t` via `tensorflow_probability.substrates.jax`. Unfortunately, TFP-on-JAX is no longer actively maintained and is incompatible with recent JAX versions, so that path is no longer viable.

## Request

Add `jax.scipy.special.owens_t(h, a)` as a native special function, matching the `scipy.special.owens_t` interface.

## Motivation

Owen's T is required for an analytic Skew Normal CDF:

```
CDF(x; ξ, ω, α) = Φ((x − ξ) / ω) − 2 T((x − ξ) / ω, α)
```

This is needed for probabilistic modeling with Skew Normal likelihoods in JAX — without a native, JIT-compatible `owens_t`, there is no clean path to an analytic Skew Normal CDF.

## Implementation notes

TFP's existing implementation (`tensorflow_probability/python/math/special.py`) is a useful reference and translates fairly directly to JAX:

- It implements the full **Patefield & Tandy (2000)** algorithm (same as SciPy)
- The `(h, a)` plane is partitioned into **18 regions**, dispatched via `tf.where` chains — already branchless and vectorized, so it maps cleanly to `jnp.where`
- Six series expansion methods are used, each tuned for numerical stability in its region
- Custom VJP/JVP are registered because the region dispatch is not cleanly differentiable through autograd; the analytical partials are closed-form:
  - `∂T/∂h = −exp(−h²/2) · erf(ah/√2) / (2√(2π))`
  - `∂T/∂a = exp(−½(a²+1)h²) / (2π(a²+1))`

The main porting work is replacing `tf.math.*` / `tf.where` with `jnp.*` / `jnp.where`, and replacing TFP's custom gradient machinery with `jax.custom_vjp` / `jax.custom_jvp`. I would be happy to contribute an implementation.

## References
- Original request: #12888
- TFP implementation (reference): [`tensorflow_probability/python/math/special.py`](https://github.com/tensorflow/probability/blob/f05a4bbbdd34d2edb199b8ce8d0b755dacbad3b7/tensorflow_probability/python/math/special.py)
- SciPy: `scipy.special.owens_t`
- Patefield & Tandy (2000): http://www.jstatsoft.org/v05/i05/paper

- [x] Check for duplicate requests.
